### ISSUE TITLE
Do not insert blank lines between list items

### ISSUE THREAD

**Issue by mkpaz** [marktext/marktext#2286]
It's not a question, neither a bug nor a feature request, but rather undesired/questionable behaviour. Typora has exactly the same issue, please see full explanation here - https://github.com/typora/typora-issues/issues/2278. In short, when code block or paragraph is added to a list, MarkText inserts blank line between each list item no matter that "Preferred loose list item" option is disabled.

I just describe my use case. I use markdown to keep and manage my tasks. I prefer to keep task lists dense (tight), because in that case it's much easer to read, but it's very often when task need an additional explanation.

```
- [ ] foo
- [ ] bar
- [ ] baz

  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Aliquam consequat placerat odio,condimentum tristique magna
  accumsan et. Maecenas id felis eu lectus sodales faucibus
  nec ac ex.

- [ ] tick

  - [ ] subtask1
  - [ ] subtask2

- [ ] tock
```

This is perfectly valid syntax, but when MarkText engine detects that list is loose it inserts additional blank lines between list items, thus making my task list two times bigger.

**Comment by trev-dev** [comment]
I agree with this, only, if it could be a _setting_, that would be even better. While I prefer no extra white space in lists, some people might prefer them.

**Comment by ivy-rew** [comment]
The seeting seems available today: 
Menu > File > Preferences > Markdown > Untoggle "List: Prefer loose list items"
Disabling it; I get the "normal" list behavior back, not adding extra new-lines.
![MarkText_002](https://github.com/marktext/marktext/assets/15085693/dd4b7bfc-8a4a-4235-804a-7f351ae8cccb)

**Comment by Jocs** [comment]
Thank you for the detailed report and the Typora comparison!

The behavior you describe — blank lines inserted between all items of a list once any item contains a multi-paragraph block — is correct per the [CommonMark specification](https://spec.commonmark.org/0.31.2/#loose), which treats looseness as a whole-list property: if any item is followed by a blank line the entire list is considered loose and every item is rendered/serialized with blank-line separation.

Since this report was filed, MarkText has added a preference that addresses the common case: **Menu → File → Preferences → Markdown → "List: Prefer loose list items"**. Disabling this option causes newly created lists to default to tight (no blank lines). A contributor confirmed this in the comments in December 2023.

For existing files that are already parsed as loose (because they contain multi-paragraph items), the `loose` flag is read directly from the source markdown by the parser, so tight items in the same list will still round-trip with blank-line separation — this is a fundamental constraint of the CommonMark spec, not a MarkText bug.

Given that the preference setting covers the main use-case and the remaining behavior is spec-mandated, we're closing this issue. If you'd like per-item looseness control as a first-class feature (which would require deviating from the spec), please open a new feature request so it can be tracked separately.
